ferrocene/doc/internal-procedures/src/testing-other-targets.rst

```diff
@@ -0,0 +1,218 @@
+          arch: "aarch64"
+          digest: "sha256:fb39312ffd2b47b97eaef6ff197912eaa3e0a215eb3eecfbf2a24acd96ee1125"
+        - location: "https://cloud-images.ubuntu.com/releases/24.10/release/ubuntu-24.10-server-cloudimg-arm64.img"
+          arch: "aarch64"
+        mounts:
+        - location: "~"
+        - location: "/tmp/lima"
+        writable: true
+        ssh:
+        forwardAgent: true
+        cpus: 8
+        memory: "8GiB"
+        mountTypesUnsupported: ["9p"]
+    EOF
+
+Start the guest:
+
+.. code-block:: bash
+    
+    limactl start ferrocene
+
+
+Shell into the guest:
+
+.. code-block:: bash
+    
+    limactl shell ferrocene
+
+You can also point `Visual Studio Code <https://code.visualstudio.com/docs/remote/ssh>`_ at it
+using `these steps <https://github.com/lima-vm/lima/discussions/1890#discussioncomment-7221563>`.
+
+
+:ref:`x86_64-pc-windows-msvc`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Setup WSL2, if you don't have it:
+
+.. code-block:: bash
+
+    wsl --install --distribution Ubuntu-24.04
+
+Ensure `nestedVirtualization` is set in the guest ``/etc/wsl.conf``, here is an example
+configuration:
+
+.. code-block::
+
+    [user]
+    default=ana
+
+    [boot]
+    systemd=true
+
+    [wsl2]
+    nestedVirtualization=true
+    
+If you changed your configuration, make sure to restart the environment with ``wsl --shutdown``.
+
+Shell into the guest:
+
+.. code-block:: bash
+    
+    wsl
+    
+You can also point `Visual Studio Code <https://code.visualstudio.com/docs/remote/wsl-tutorial>`_ at it.
+
+Target Procedures
+-----------------
+
+Currently bare metal targets have a similar procedure for testing.
+
+:ref:`aarch64-unknown-none`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. Note::
+    
+    In a :target:`aarch64-unknown-linux-gnu` environment (such as a Lima guest on :ref:`aarch64-apple-darwin`)
+    you may simply skip to the final step, running the tests, as no QEMU is needed.
+
+Install the necessary packages:
+
+.. code-block:: bash
+
+    sudo apt install gcc-aarch64-linux-gnu qemu-system-aarch64
+
+If you don't already have a ``/usr/share/binfmts/qemu-aarch64`` file, create one:
+
+.. code-block:: bash
+
+    package qemu-aarch64
+    interpreter /usr/bin/qemu-aarch64-static
+    magic \x7fELF\x02\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02\x00\xb7\x00
+    mask \xff\xff\xff\xff\xff\xff\xff\x00\xff\xff\xff\xff\xff\xff\xff\xff\xfe\xff\xff\xff
+    credentials no
+    preserve no
+    fix_binary no
+    
+Then make sure it's imported:
+
+.. code-block:: bash
+    
+   sudo update-binfmts --import qemu-aarch64
+
+You can now run the tests:
+
+.. code-block:: bash
+
+    ./x test --stage 1 --target aarch64-unknown-ferrocenecoretest library/core
+
+:target:`thumbv7em-none-eabihf` & :target:`thumbv7em-none-eabi`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Install the necessary packages:
+
+.. code-block:: bash
+
+    sudo apt install gcc-arm-none-eabi qemu-system-arm
```

Comment:
```suggestion
    sudo apt install gcc-arm-none-eabi qemu-system-arm binfmt-support
```

`update-binfmts` needs to be installed on Ubuntu 24.04 before we can run this.

Reply:
`cc` is covered in setup-local-env, previous feedback asked me to move it there.